src/stan/services/sample/hmc_nuts_dense_e_adapt.hpp

```diff
@@ -0,0 +1,126 @@
+#ifndef STAN_SERVICES_SAMPLE_HMC_NUTS_DENSE_E_ADAPT_HPP
+#define STAN_SERVICES_SAMPLE_HMC_NUTS_DENSE_E_ADAPT_HPP
+
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <stan/callbacks/interrupt.hpp>
+#include <stan/callbacks/writer.hpp>
+#include <stan/mcmc/fixed_param_sampler.hpp>
+#include <stan/services/error_codes.hpp>
+#include <stan/mcmc/hmc/nuts/adapt_dense_e_nuts.hpp>
+#include <stan/services/util/run_adaptive_sampler.hpp>
+#include <stan/services/util/rng.hpp>
+#include <stan/services/util/initialize.hpp>
+#include <vector>
+
+namespace stan {
+  namespace services {
+    namespace sample {
+
+      /**
+       * Runs HMC with NUTS with dense Euclidean
+       * metric without adapatation.
+       *
+       * @tparam Model Model class
+       *
+       * @param model Input model to test (with data already instantiated)
+       * @param init var context for initialization
+       * @param random_seed random seed for the pseudo random number generator
+       * @param chain chain id to advance the pseudo random number generator
+       * @param init_radius radius to initialize
+       * @param num_warmup Number of warmup samples
+       * @param num_samples Number of samples
+       * @param num_thin Number to thin the samples
+       * @param save_warmup Indicates whether to save the warmup iterations
+       * @param refresh Controls the output
+       * @param stepsize initial stepsize for discrete evolution
+       * @param stepsize_jitter uniform random jitter of stepsize
+       * @param max_depth Maximum tree depth
+       * @param delta
+       * @param gamma
+       * @param kappa
+       * @param t0
+       * @param init_buffer
+       * @param term_buffer
+       * @param window
+       * @param interrupt Callback for interrupts
+       * @param message_writer Writer for messages
+       * @param error_writer Writer for errors
+       * @param init_writer Writer callback for unconstrained inits
+       * @param sample_writer Writer for draws
+       * @param diagnostic_writer Writer for diagnostic information
+       * @return error code; 0 if no error
```

Comment:
I think with optimization you were referring to stan::services::error_codes::OK for teh error codes.